src/components/admin/agency/agency-profile-form.js

```diff
@@ -1,30 +1,78 @@
-import React, { useRef, useCallback } from "react";
-import { Grid, TextField, Button, Box, Divider } from "@material-ui/core";
+import React, { useRef, useCallback, useEffect, useState } from "react";
+
 import { Formik } from "formik";
 import * as yup from "yup";
+
+import { Grid, TextField, Button, Box, Divider, CircularProgress, Backdrop } from "@material-ui/core";
+
+import AuthClient from "../../../api/auth-client";
+import FileClient from "../../../api/file-client";
+
+import { AuthContext } from "../../../store/context";
 import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
 import LogoUpload from "../agency/logo-upload";
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles((theme) => ({
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+}));
 
 const AgencyProfileSchema = yup.object().shape({
-  phone: yup.number().typeError("Phone must be a number").required("Required"),
+  phoneNumber: yup.number().typeError("Phone must be a number").required("Required"),
 });
 
 export default function AgencyProfileForm() {
+  const formik = useRef(null);
   const logoInput = useRef(null);
   const logoFormik = useRef(null);
 
-  function handleBrowseLogoClick() {
+  const classes = useStyles();
+  const authContext = JSON.parse(localStorage.getItem("AuthContext"));
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => { 
+    const getAgency = async () => {
+      await AuthClient.getAgencyProfile(authContext.accessToken).then((response) => {
+        if(response.status === 200){
```

Comment:
you can skip status check, because axios response promise rejects if there is a http error. you can handle errors by catch(callback) method